include/utilities/TimestampEstimator.hpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * @file TimestampEstimator.hpp TimestampEstimator Class
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#ifndef UTILITIES_INCLUDE_UTILITIES_TIMESTAMPESTIMATOR_HPP_
+#define UTILITIES_INCLUDE_UTILITIES_TIMESTAMPESTIMATOR_HPP_
+
+#include "utilities/TimestampEstimatorBase.hpp"
+#include "utilities/Issues.hpp"
+
+#include "utilities/WorkerThread.hpp"
+
+#include <atomic>
+#include <memory>
+#include <mutex>
+#include <thread>
+
+namespace dunedaq {
+namespace utilities {
+
+/**
+ * @brief TimestampEstimator is an implementation of
+ * TimestampEstimatorBase that uses TimeSync messages from an input
+ * queue to estimate the current timestamp
+ **/
+class TimestampEstimator : public TimestampEstimatorBase
+{
+public:
+  TimestampEstimator(uint32_t run_number, uint64_t clock_frequency_hz);
+
+  explicit TimestampEstimator(uint64_t clock_frequency_hz); // NOLINT(build/unsigned)
+
+  virtual ~TimestampEstimator();
+
+  uint64_t get_timestamp_estimate() const override { return m_current_timestamp_estimate.load(); }
+
+  void add_timestamp_datapoint(uint64_t daq_time, uint64_t system_time);
+
+  template <class T>
+  void timesync_callback(T& tsync);
+
+  uint64_t get_received_timesync_count() { return m_received_timesync_count.load(); }
+private:
+  void estimator_thread_fn(std::atomic<bool>& running_flag);
```

Comment:
Could the declaration of the estimator_thread_fn() method be removed?  It doesn't seem to be used anywhere.  (and then we could also remove the #include of WorkerThread and <thread>)